### ISSUE TITLE
 16bit program cannot find haxm

### ISSUE THREAD

**Issue by SkillGodAk** [leecher1337/ntvdmx64#236]
I installed HAXM versions
But executing the program I want to run shows that haxm is not installed
I'm sure hyper-v is off and vt in bios is on
Use cmd to check and make sure it is open. Where is the problem?

![01](https://github.com/leecher1337/ntvdmx64/assets/104513085/d51744f1-26a7-45cd-b0b5-f75b4d8c3cea)

**Comment by leecher1337** [comment]
Wrong service. You mustn't run `IntelHaxm` service! It has to be `IntelHaxmNtvdm` shipped with HAXM build. To be safe, ensure that original IntelHaxm driver doesn't run.

https://github.com/leecher1337/ntvdmx64/blob/0f7f33e295024b81681d506cc86b8606960ef713/ntvdmpatch/doc/haxm.txt#L93

**Comment by SkillGodAk** [comment]
IntelHaxm and IntelHaxmNtvdm, which have been built with HAXM in the forum, will not jump out of the error when executing the program, but there is no response when executing the program. What is the problem? This software is cracked software, and the cracked file needs to be placed under the c:\system to run. I I don't know if it affects

**Comment by theliz12x** [comment]
Hi i am tring to install the HAXM version of NTVLDMX64 becouse the CCPU seems quiet slow with softwares  but i am a bit in trouble. i've installre the Version 7.8.0 from https://github.com/intel/haxm 
at first attempt to launch an application (simpe EDIT) i've received an error
"cannot setup exception bitmap for "BOPping" you are running our custom haxm version... close"
then i've checkd all the requirement into the readme.txt (checktool.exe show all requirements present) 
i suppose i am using the official HAXM downloaded from 
The latest HAXM release for Windows and macOS hosts are available 
https://github.com/intel/haxm/releases/latest

is that my fault? 
(i've deleted also the present service called IntelHAXMNtvdm that now is not present anymore)
in the project https://github.com/leecher1337/haxm there is not haxm release versions that i can download?
i've deleted also the service called IntelHAXM and now i receive the message that no Intelhaxm is installed

**Comment by leecher1337** [comment]
Hi,
Unfortunately, leaked NVIDIA code signing certficate was blocked on most Windows installations by now and therefore isn't useful any longer, so running Windows in test signing mode using the old driver seems to be the only way, as can be seen bey the error message you got.
It's mandatory to use my custom IntelHAXM-version, as Intel refused to integrate my improvements in their main driver, unfortunately (I tried many times to persuade them, but to no avail, even though, these are only enhancements fully compatible with the main driver).
Thus, try test signing mode with older IntelHaxm driver with test signed certificate now that I reverted back now in last commit.

**Comment by theliz12x** [comment]
> Hi, Unfortunately, leaked NVIDIA code signing certficate was blocked on most Windows installations by now and therefore isn't useful any longer, so running Windows in test signing mode using the old driver seems to be the only way, as can be seen bey the error message you got. It's mandatory to use my custom IntelHAXM-version, as Intel refused to integrate my improvements in their main driver, unfortunately (I tried many times to persuade them, but to no avail, even though, these are only enhancements fully compatible with the main driver). Thus, try test signing mode with older IntelHaxm driver with test signed certificate now that I reverted back now in last commit.

i've tried your custom version (i did it in the way i've copied the "patched" IntelHaxm.sys ) but with the same result, "not installed" and i see that the service is not started i suppose becouse of the "signed certificate issue". if i use your patched version shuold i cross the problem or i missunderstood?

**Comment by theliz12x** [comment]
i've tried your custom version (i did it in the way i've copied the "patched" IntelHaxm.sys ) but with the same result, "not installed" and i see that the service is not started i suppose becouse of the "signed certificate issue". if i use your patched version shuold i cross the problem or i missunderstood?
 
Da: leecher1337 ***@***.***> 
Inviato: giovedì 29 febbraio 2024 12:58
A: leecher1337/ntvdmx64 ***@***.***>
Cc: theliz12x ***@***.***>; Comment ***@***.***>
Oggetto: Re: [leecher1337/ntvdmx64] 16bit program cannot find haxm (Issue #236)
 
Hi,
Unfortunately, leaked NVIDIA code signing certficate was blocked on most Windows installations by now and therefore isn't useful any longer, so running Windows in test signing mode using the old driver seems to be the only way, as can be seen bey the error message you got.
It's mandatory to use my custom IntelHAXM-version, as Intel refused to integrate my improvements in their main driver, unfortunately (I tried many times to persuade them, but to no avail, even though, these are only enhancements fully compatible with the main driver).
Thus, try test signing mode with older IntelHaxm driver with test signed certificate now that I reverted back now in last commit.

**Comment by leecher1337** [comment]
So, did you take this version (that I downgraded 2 hours ago, you need the new .cer and the new .sys and add the .cer to the store): 
https://github.com/leecher1337/ntvdmx64/tree/master/ntvdmpatch/haxm

And did you enable test signing mode?

```
bcdedit /set testsigning on
certutil -addstore "TrustedPublisher" haxm\IntelHaxm.cer
```

In testsigning mode, it shouldn't be too picky about the certificate as long as it is installed in the store of trusted publishers.

**Comment by theliz12x** [comment]
Hi @leecher1337 , i did it , now with this last version the service is started and then i do not receive anymore the message about "haxm not installed" but i still have another issue an error message if i try to use edit.com
"16 bit application not supported" , etc etc

**Comment by theliz12x** [comment]
it works, following other issues i were checking file folders, i had to take ownership of syswow64 so that the installation process could replace corretly all the files so that maybe this can help others.
@leecher1337 it will works until 23.03.2026 (the date of certificare expiration?)   there is no way to let the service starts without need of the signed certs?
thank you now the speed of TP compiler is 2 seconds for 1  million lines, the same of an old I7 of 10 years ago :)

**Comment by leecher1337** [comment]
@theliz12x Glad to hear that it worked for you now! 👍 
The issues with Syswow64 ownership is strange, never encountered this on a system yet, I thought admin user that is needed to install it will have permissions to write ot syswow64 anyway. 
Normally the installer does `takeown /f` only for wow32.dll and user.exe, as these are files that are already present from Windows and need to be replaced, but the other stuff normally isn't there on a new system so it should be able to write it.
Anyway, good that you documented it here! 
If anyone else is also experiencing this problem, informations on how to reproduce this are welcome.

Regarding the signature expiration date of the certificiate: This is not a problem. It is just necessary that the certificate is valid at the time of signing the driver.

Currently I don't have a way to allow the driver to be loaded outside of test signing mode, due to now having contact to a company that posesses a driver signing certificate and is willing to pass the driver through Microsoft's signing process.
If anyone is able to sign the driver, I would be delighted!

There are hacks like "Bring your own vulnerable driver":
https://github.com/hfiref0x/KDU
This requires a signed vulnerable driver to chain-load your driver, but it's a hack and these drivers frequently get blacklisted too, so it's also only a temporary fix.  So I guess it's not much better than the leaked certificate I used before.

Your use-case is the one that haxm support was developed for - speeding up textmode applications. Good to hear that it gets used

**Comment by theliz12x** [comment]
Yes i will use it. Great job!!! i still use some old compilers on 32b machines. It works very well, i will let you know if any issue can occur. Rightnow i've found only that problem with LPT connected with net use as i told you in another post. DOSPRN can be a workaround i will try it. Thank you

**Comment by leecher1337** [comment]
Did you use the most recent repository for building?
Because driver got reverted in https://github.com/leecher1337/ntvdmx64/commit/eb24d233ce08e31cf3e5dfb3c4b8e29733489b59
to the testsigning mode version. Did you follow procedure to enable testsigning mode as specified in the docs?

**Comment by killerkirb** [comment]
i wasn't using the most recent i think anyway i decided to use ccpu this time thx for helping
